lib/worker.js

```diff
@@ -283,10 +283,15 @@ class AssetComputeWorker {
             if (await this.shouldPostProcess(rendition)) {
 
                 if (rendition.instructions.watermark && rendition.instructions.watermark.watermarkContent !== undefined && rendition.instructions.watermark.watermarkContent !== null) {
-                    rendition.instructions.watermark = await getWatermark(
-                        rendition.instructions.watermark,
+                    const watermarkObject = Object.assign({
+                        url: rendition.instructions.watermark.watermarkContent
+                    }, rendition.instructions.watermark);
+
+                    rendition.instructions.watermark = await getSource(
+                        watermarkObject,
                         this.directories.in,
-                        this.options.disableSourceDownload
+                        this.options.disableSourceDownload,
+                        'watermark'
```

Comment:
I find it confusing that 'watermark' here is the file's basename. Reading this without knowing the download code, I would think 'watermark' is an enum or constant value to indicate we are actually downloading a watermark.

Reply:
Can't we actually put this information in the watermark object?